### ISSUE TITLE
not match between CheX_Dataset and model with weights="densenet121-res224-chex"

### ISSUE THREAD

**Issue by catfish132** [mlmed/torchxrayvision#107]
I found that the summary of CheX_Dataset has 13 classes of disease ,but in the head of model.py ，model with weights="densenet121-res224-chex" provide 11 classes of disease, I think this is not matched. And as same as "mimic_ch" and "mimic_nb".
this means the model will provide 7 useless dimensions for each sample. Is there any kind of disease missing?

**Comment by ieee8023** [comment]
'Plural Other' and 'Support Devices' were not included because they didn't overlap with the list in `xrv.datasets.default_pathologies` which was just an arbitrary mix of common pathologies. The model predicts 0.5 because it was not trained on that pathology. For each model where weights are loaded you can access the `model.pathologies` field to see what pathologies it was trained to predict.

```python3
model = xrv.models.DenseNet(weights="densenet121-res224-chex")

x = torch.ones(1,1,224,224)
preds = model(x).detach().numpy().tolist()[0]

for patho, pred, in zip(model.pathologies, preds):
    print(patho, pred)

Atelectasis 0.8239850997924805
Consolidation 0.8161774277687073
 0.5
Pneumothorax 0.582581639289856
Edema 0.8048802614212036
 0.5
 0.5
Effusion 0.8671234846115112
Pneumonia 0.6745733022689819
 0.5
Cardiomegaly 0.7989252805709839
 0.5
 0.5
 0.5
Lung Lesion 0.7476078271865845
Fracture 0.6972948312759399
Lung Opacity 0.8870707750320435
Enlarged Cardiomediastinum 0.7584663033485413
```

**Comment by catfish132** [comment]
If I want to pre-train a model on MIMIC with 14 labels what should I do ?

**Comment by ieee8023** [comment]
I think there are only 13 labels in MIMIC: https://github.com/mlmed/torchxrayvision/blob/master/torchxrayvision/datasets.py#L989

You can train a new model here: https://github.com/mlmed/torchxrayvision/blob/master/scripts/train_model.py

Set the dataset to "mimic_ch" https://github.com/mlmed/torchxrayvision/blob/master/scripts/train_model.py#L92

**Comment by catfish132** [comment]
yeah，I tried and got this output:

datas_names ['mimic_ch']
{'Support Devices', 'Pleural Other'} will be dropped
Infiltration doesn't exist. Adding nans instead.
Emphysema doesn't exist. Adding nans instead.
Fibrosis doesn't exist. Adding nans instead.
Pleural_Thickening doesn't exist. Adding nans instead.
Nodule doesn't exist. Adding nans instead.
Mass doesn't exist. Adding nans instead.
Hernia doesn't exist. Adding nans instead.
train_dataset.labels.shape (194651, 18)
test_dataset.labels.shape (48667, 18)

I think this means the trained model will have only 11 useful output , how can I add 'Support Devices', 'Pleural Other' back to the dataset?

**Comment by ieee8023** [comment]
This is the line that is doing it: https://github.com/mlmed/torchxrayvision/blob/master/scripts/train_model.py#L140

**Comment by catfish132** [comment]
Hello ，the problem above is solved. However I found this row of the code:
    #disable data augs
    test_dataset.data_aug = None (in train_modal.py @157 line)
if I not comment this row,the program throw a error and terminate.So is it ok that comment this row when I train a resnet101 model . For more detials I am using MIMIC-ch dataset .

**Comment by catfish132** [comment]
Hello ，the problem above is solved. However I found this row of the code:
#disable data augs
test_dataset.data_aug = None (in train_modal.py @157 line)
if I not comment this row,the program throw a error and terminate.So is it ok that comment this row when I train a resnet101 model ? For more detials I am using MIMIC-ch dataset .

**Comment by catfish132** [comment]
By the way, I try to train a resnet101 with batchsize=8 on 2080ti GPU , and it cost 7000+MB memory already. This is far beyond my expectations, I think it should cost less memory. Is there any problem I may  ignored?

**Comment by catfish132** [comment]
By the way, I try to train a resnet101 with batchsize=8 on 2080ti GPU , and it cost 7000+MB memory already. This is far beyond my expectations, I think it should cost less memory. Is there any problem I may  ignored?

**Comment by ieee8023** [comment]
Ah yes, sorry about that. Since that script was made the subset dataset doesn't let the user modify the transform field. It is ok to comment it out. The data aug will happen in the validation and test set datasets because they are created with the data aug. The alternative is to load the dataset twice (once with and once without the data augs) and then use the one without data aug for testing. I'll remove that line from the script.

**Comment by ieee8023** [comment]
As for the memory you can set the resolution to something smaller like 224. In that script now it is 512.
